### PR TITLE
apulse: 0.1.10 -> 0.1.11, add pressureaudio

### PR DESCRIPTION
**Pull request by oxij**
###### Motivation for this change

Living without PulseAudio.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by oxij** [comment]
I tested `mpv` and `cmus`, both work fine with this and with plain ALSA disabled. `firefox` gets "permission denied" errors, probably because of sandboxing. More debugging needed there.
Sharing in case anyone else wants to play with this.

**Comment by 7c6f434c** [comment]
Firefox 58 is mentioned on the `apulse` homepage, there is an about:config option to loosen the sandbox and allow `ioctl`.

**Comment by oxij** [comment]
Ah, thanks!

File: pkgs/misc/apulse/pressureaudio.nix [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  configurePhase = ''
+    echo "Doing nothing."
+  '';
+
+  buildPhase = ''
+    echo "Building zero lines of code."
+  '';
+
+  installPhase = ''
+    echo "Coping libraries."
```
Comment (7c6f434c) [inline pkgs/misc/apulse/pressureaudio.nix]: Either use `:` for empty phases and reduce verbosity here, or fix the typo: ‘‘Coping’’ → ‘‘Copying’’

File: pkgs/misc/apulse/pressureaudio.nix [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    mkdir -p $out/lib
+    ls ${apulse}/lib/apulse $out/lib
+    cp -a ${apulse}/lib/apulse/* $out/lib/
```
Comment (7c6f434c) [inline pkgs/misc/apulse/pressureaudio.nix]: Any reason for `cp -a` instead of `ln -s`?

File: pkgs/misc/apulse/pressureaudio.nix [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  meta = apulse.meta // {
+    description = "libpulseaudio without any sound daemons over pure ALSA";
+    longDescription = ''
+      apulse (${apulse.meta.homepage}) implements most of libpulseaudio
+      API over pure ALSA in 5% LOC of the original PulseAudio.
+
+      But apulse is made to be used as a wrapper that substitutes its
+      replacement libs into LD_LIBRARY_PATH. The problem with that is
+      that you have to link against the original libpulseaudio, which
+      means you have to build it (and risk executing it eventually).
```
Comment (7c6f434c) [inline pkgs/misc/apulse/pressureaudio.nix]: Maybe remove this paragraph (and the last one)? As it is, `libpulseaudio` packages in Nix does not build any executables, and I am not sure there are any risks in failed attempts to connect by PA client library.

**Review by 7c6f434c — DISMISSED** [review]

**Comment by oxij** [comment]
> Either use `:` for empty phases and reduce verbosity here, or fix the typo: ‘‘Coping’’ → ‘‘Copying’’

Oops, that and the correct .pc files were lost in the git stash. Fixed. Thanks.

> Any reason for `cp -a` instead of `ln -s`?

I tried, but it gets ugly fast as the original has symlinks you have to
preserve too. It's gonna be rehardlinked by nix-store store optimization
anyway. (Header files too, btw.)

> Maybe remove this paragraph (and the last one)? As it is, `libpulseaudio` packages in Nix does not build any executables, and I am not sure there are any risks in failed attempts to connect by PA client library.

Fixed.

**Pull request merged**